src/util/ecdsa.rs

```diff
@@ -413,6 +415,61 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+/// An ECDSA signature with the corresponding hash type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EcdsaSig {
+    /// The underlying ECDSA Signature
+    pub sig: secp256k1::Signature,
+    /// The corresponding hash type
+    pub hash_ty: EcdsaSigHashType,
+}
+
+impl EcdsaSig {
+
+    /// Deserialize from slice
+    pub fn from_slice(sl: &[u8]) -> Result<Self, EcdsaSigError> {
+        let (hash_ty, sig) = sl.split_last()
+            .ok_or(EcdsaSigError::InvalidEcdsaSig)?;
+        let hash_ty = EcdsaSigHashType::from_u32_standard(*hash_ty as u32)
+            .map_err(|_| EcdsaSigError::NonStandardSigHashType(*hash_ty))?;
+        let sig = secp256k1::Signature::from_der(sig)
+            .map_err(|_| EcdsaSigError::InvalidEcdsaSig)?;
```

Comment:
Why throw out the errors?

Reply:
The earlier versions of the PR did capture the underlying errors. I did this in order to 
- Avoid all but one unreachable error states in `secp256k1::Error`. In this scenario, I did check that only one variant of error was being returned in the corresponding secp functions. 
- It would have required to me have another return error type for empty signature. I don't know if the end-user really cares about this error type when the signature is empty or if the signature is malformed. I want to avoid creating unnecessary error enums that are not useful, but it makes matching with them downstream even harder. 

I was recently convinced on rust discord by someone else that I should always capture because in case upstream secp changes error reporting, we might want to capture it. Changing it back to the way it was previously